### PR TITLE
Simplify the flutter_web_plugins plugin registration API.

### PR DESCRIPTION
**Pull request by Hixie**
In principle this is backwards compatible; I just merged all the classes into one and deprecated everything that became redundant as a result.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @hterkelsen

File: packages/flutter_web_plugins/lib/src/plugin_registry.dart [hunk 1]
```diff
@@ -8,93 +8,106 @@ import 'dart:ui' as ui;
...
-  ///       'com.my_plugin/my_plugin',
-  ///       const StandardMethodCodec(),
-  ///       registrar.messenger,
-  ///     );
-  ///     final MyPlugin instance = MyPlugin();
-  ///     channel.setMethodCallHandler(instance.handleMethodCall);
-  ///   }
-  ///   // ...
-  /// }
+  /// ui.webOnlySetPluginHandler(webPluginRegistar.handleFrameworkMessages);
```
Comment (hterkelsen) [inline packages/flutter_web_plugins/lib/src/plugin_registry.dart:108]: nit: `webPluginRegistar` -> `webPluginRegistrar`

File: packages/flutter_tools/lib/src/isolated/resident_web_runner.dart [hunk 2]
```diff
@@ -691,7 +691,7 @@ class _ResidentWebRunner extends ResidentWebRunner {
         'typedef _NullaryFunction = dynamic Function();',
         'Future<void> main() async {',
         if (hasWebPlugins)
-          '  registerPlugins(webPluginRegistry);',
+          '  registerPlugins(webPluginsRegistrar);',
```
Comment (hterkelsen) [inline packages/flutter_tools/lib/src/isolated/resident_web_runner.dart:694]: `webPluginsRegistrar` -> `webPluginRegistrar`

File: packages/flutter_tools/lib/src/plugins.dart [hunk 3]
```diff
@@ -789,11 +789,11 @@ import 'package:{{name}}/{{file}}';
...
 
 // ignore: public_member_api_docs
-void registerPlugins(PluginRegistry registry) {
+void registerPlugins(Registrar registrar) {
 {{#plugins}}
-  {{class}}.registerWith(registry.registrarFor({{class}}));
+  {{class}}.registerWith(registrar);
 {{/plugins}}
-  registry.registerMessageHandler();
+  ui.webOnlySetPluginHandler(registrar.handleFrameworkMessages);
```
Comment (hterkelsen) [inline packages/flutter_tools/lib/src/plugins.dart:796]: `handleFrameworkMessages` -> `handleFrameworkMessage`

**Review by harryterkelsen — APPROVED** [review]
LGTM if generated code is fixed

File: packages/flutter_web_plugins/lib/src/plugin_registry.dart (hunk 1 above)
Comment (hterkelsen) [reply packages/flutter_web_plugins/lib/src/plugin_registry.dart:108]: Also `handleFrameworkMessages` -> `handleFrameworkMessage`

File: packages/flutter_tools/lib/src/plugins.dart (hunk 3 above)
Comment (Hixie) [reply packages/flutter_tools/lib/src/plugins.dart:796]: turns out i also need to change the import of "dart:ui" to "as ui".
(Any idea why we had the import before if it was unused?)

**Pull request merged**